src/Administration/Resources/app/administration/src/module/sw-product/page/sw-product-detail/index.js

```diff
@@ -1216,5 +1248,55 @@ export default {
                 });
             });
         },
+
+        async initializeProductUnits() {
+            const store = Shopware.Store.get('swProductDetail');
+            const preferredUnits = await this.getPreferredUnitsConfig();
+            if (preferredUnits) {
+                store.setLengthUnit(preferredUnits.length);
+                store.setWeightUnit(preferredUnits.weight);
+                return;
+            }
+
+            this.measurementSystemConfig = await this.getMeasurementSystemConfig();
+            if (!this.measurementSystemConfig['core.measurementSystem.lengthUnitId']
+                || !this.measurementSystemConfig['core.measurementSystem.weightUnitId']) {
+                return;
+            }
+
+            const units = await this.measurementDisplayUnitRepository.search(
+                this.measurementDisplayUnitCriteria,
+            );
+
+            if (!units?.length) {
+                return;
+            }
+
+            const displayUnits = units.reduce((acc, unit) => {
+                acc[unit.type] = unit.shortName;
+                return acc;
+            }, {});
+
+            store.setLengthUnit(displayUnits.length);
+            store.setWeightUnit(displayUnits.weight);
+        },
+
+        getMeasurementSystemConfig() {
+            return this.systemConfigApiService.getValues('core.measurementSystem');
+        },
+
+        async getPreferredUnitsConfig() {
+            const response = await this.userConfigService.search(['measurement.preferredUnits']);
+            return response.data['measurement.preferredUnits'];
+        },
+
+        savePreferredUnitsConfig() {
+            return this.userConfigService.upsert({
+                'measurement.preferredUnits': {
+                    'length': this.lengthUnit,
+                    'weight': this.weightUnit,
+                },
```

Comment:
'measurement.preferenceUnit': {
                    'lengthUnit': this.lengthUnit,
                    'weightUnit': this.weightUnit,
                },